examples/player/simid_player.js

```diff
@@ -609,16 +609,69 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+
+    //If the request duration is longer than the ad duration, the ad extends for the requested amount of time
+    if (this.requestedDuration_ != UNLIMITED_DURATION) {
+      const durationChangeMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+      setTimeout(() => {
+        this.stopAd(StopCode.CREATIVE_INITIATED);
+      }, durationChangeMs);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    const originalDuration = this.requestedDuration_;
+    this.requestedDuration_  = incomingMessage.args['duration'];
```

Comment:
imo this would read a little easier if you only assigned the new duration to `this.requestedDuration_` in the else branch, rather than storing the original and resetting

Reply:
if I do this, then in the case where the user inputs a negative value, the ad will stop immediately and not show the entire ad (which is the desired functionality)

Reply:
Can you explain why that would be the case?

if you change this so that its something like

```
const newDuration = incomingMessage.args['duration'];
if (newDuration != UNLIMITED && newDuration < 0) {
  this.simidProtocol.reject(...);
} else {
  this.requestedDuration_ = newDuration;
  ...
}
```

I don't think I see how the end result would be any different

Reply:
Oh I misunderstood what you were referring to when you said new duration. I thought you meant to still use the this.reqeuestedDuration_ in the if statement which wouldn't work. This will work and I will change it